src/core/comm/coll.cc

```diff
@@ -264,9 +264,9 @@ int collInit(int argc, char* argv[])
   CHECK_MPI(MPI_Comm_get_attr(MPI_COMM_WORLD, MPI_TAG_UB, &tag_ub, &flag));
   assert(flag);
   mpi_tag_ub = *tag_ub;
-  mpi_comms.clear();
+  assert(mpi_comms.empty() == true);
```

Comment:
isn't `empty` already returning a boolean value? what's the point of this comparison?